packages/x-components/src/x-modules/history-queries/store/actions/update-history-queries-with-search-response.action.ts

```diff
@@ -31,7 +31,7 @@ export const updateHistoryQueriesWithSearchResponse: HistoryQueriesXStoreModule[
         const historyQuery = state.historyQueries[indexOfHistoryQuery];
         const isCurrentSessionHistoryQuery = historyQuery.timestamp > state.sessionTimeStampInMs;
         if (!isCurrentSessionHistoryQuery || historyQuery.totalResults == null || searchResponse) {
-          const filters = createHistoryQueriesFiltersList(searchResponse.request.filters);
+          const filters = createHistoryQueriesFiltersList(searchResponse.facets);
```

Comment:
This is sketchy. The search response filters aren't selected, but this isn't the actual search response, this is the search response rebuilt from the fields in the search module state.
The search module state changes because of the facets service, which is the thing that changes the filters to selected when they arrive from the search response.

This approach might not always work. This action is called from the wiring due to a search module emitter event and when that happens the filters are unselected. Is working now because the filters service happens to process the filters before the wiring of the history queries module receives the event to call the action.

It's worth checking it a little bit further and check if it's possible that this action runs before the facets service processes the filters